pkg/agent/docker_secrets_postrenderer.go

```diff
@@ -152,11 +152,12 @@ func (r *DockerSecretsPostRenderer) updatePodSpecWithPullSecrets(podSpec map[int
 	var imagePullSecrets []map[string]interface{}
 	existingNames := map[string]bool{}
 	if existingPullSecrets, ok := podSpec["imagePullSecrets"]; ok {
-		imagePullSecrets = existingPullSecrets.([]map[string]interface{})
-		for _, s := range imagePullSecrets {
-			if name, ok := s["name"]; ok {
+		for _, s := range existingPullSecrets.([]interface{}) {
```

Comment:
It's not new in this PR, but shouldn't those type conversions be using the comma, ok idiom, even if we're certain that it'll always convert... if we're wrong, it will [crash with a run-time error](https://golang.org/doc/effective_go#interface_conversions). (Apologies if I wrote the initial type conversion without the comma,ok).

Reply:
Hah - and I see this was the actual issue being addressed (runtime error due to a failed interface conversion), so yes, we really should handle that with the comma-ok, IMO.